components/style/properties/longhand/font.mako.rs

```diff
@@ -400,24 +400,120 @@ ${helpers.single_keyword("font-variant-caps",
+                _ => false
+            }
         }
     }
 
     #[derive(Debug, Clone, PartialEq)]
     #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
-    pub struct SpecifiedValue(pub specified::LengthOrPercentage);
+    pub enum SpecifiedValue {
+        Length(specified::LengthOrPercentage),
+        Keyword(KeywordSize),
+    }
+
     pub mod computed_value {
         use app_units::Au;
         pub type T = Au;
     }
 
+    /// CSS font keywords
+    #[derive(Debug, Copy, Clone, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum KeywordSize {
+        XXSmall = 0,
+        XSmall = 1,
+        Small = 2,
+        Medium = 3,
+        Large = 4,
+        XLarge = 5,
+        XXLarge = 6,
+        // This is not a real font keyword and will not parse
+        // HTML font-size 7 corresponds to this value
+        XXXLarge = 7,
+    }
+
+    pub use self::KeywordSize::*;
+
+    impl KeywordSize {
+        pub fn parse(input: &mut Parser) -> Result<Self, ()> {
+            Ok(match_ignore_ascii_case! {&*input.expect_ident()?,
+                "xx-small" => XXSmall,
+                "x-small" => XSmall,
+                "small" => Small,
+                "medium" => Medium,
+                "large" => Large,
+                "x-large" => XLarge,
+                "xx-large" => XXLarge,
+                _ => return Err(())
+            })
+        }
+    }
+
+    impl ToCss for KeywordSize {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            dest.write_str(match *self {
+                XXSmall => "xx-small",
+                XSmall => "x-small",
+                Small => "small",
+                Medium => "medium",
+                Large => "large",
+                XLarge => "x-large",
+                XXLarge => "xx-large",
+                XXXLarge => "",
+            })
+        }
+    }
+
+    impl ToComputedValue for KeywordSize {
+        type ComputedValue = Au;
+        #[inline]
+        fn to_computed_value(&self, _: &Context) -> computed_value::T {
+            // https://drafts.csswg.org/css-fonts-3/#font-size-prop
+            use values::FONT_MEDIUM_PX;
+            match *self {
+                XXSmall => Au::from_px(FONT_MEDIUM_PX) * 3 / 5,
+                XSmall => Au::from_px(FONT_MEDIUM_PX) * 3 / 4,
+                Small => Au::from_px(FONT_MEDIUM_PX) * 8 / 9,
+                Medium => Au::from_px(FONT_MEDIUM_PX),
+                Large => Au::from_px(FONT_MEDIUM_PX) * 6 / 5,
+                XLarge => Au::from_px(FONT_MEDIUM_PX) * 3 / 2,
+                XXLarge => Au::from_px(FONT_MEDIUM_PX) * 2,
+                XXXLarge => Au::from_px(FONT_MEDIUM_PX) * 3,
+            }
+        }
+
+        #[inline]
+        fn from_computed_value(_: &computed_value::T) -> Self {
+            unreachable!()
+        }
+    }
+
+    impl SpecifiedValue {
+        pub fn from_html_size(size: u8) -> Self {
```

Comment:
Probably worth a comment here referring to HTML spec's related section and mentioning
> If value is greater than 7, let it be 7.
> If value is less than 1, let it be 1.